### ISSUE TITLE
Configured ObjectMapperSupplier not instanciated if deployed to application server

### ISSUE THREAD

**Issue by JensPiegsa** [vladmihalcea/hibernate-types#309]
Based on your nice blog posts ([1](https://vladmihalcea.com/how-to-map-json-objects-using-generic-hibernate-types/), [2](https://vladmihalcea.com/how-to-map-json-collections-using-jpa-and-hibernate/), [3](https://vladmihalcea.com/hibernate-types-customize-jackson-objectmapper/)), I use hibernate-types-52 version 2.10.3 to serialize a `java.util.Map` to a PostgreSQL JSONB column:

```
	@Basic(fetch = FetchType.LAZY)
	@Column(columnDefinition = "jsonb")
	@Type(type = "jsonb")
	private Map<String, Object> customFieldData = new LinkedHashMap<>();
```

I implemented a custom `ObjectMapperSupplier` and referenced this class via `hibernate.types.jackson.object.mapper` in `hibernate-types.properties`. I registered a custom `com.fasterxml.jackson.databind.JsonSerializer` and a `com.fasterxml.jackson.databind.JsonDeserializer`.

This all together works fine in my `testcontainers` integration test, where I use `Persistence.createEntityManagerFactory(..)` to get the EntityManager.

However, trying to persist my Entity in the deployed application (WildFly 22.0.1) fails, because Jackson does not use the provided `ObjectMapper`. My class implementing `ObjectMapperSupplier` is never instantiated.

Any idea what might have gone wrong?

**Comment by vladmihalcea** [comment]
It could be a class loading issue caused by JBoss module system. I suppose the `hibernate-types.properties` file is not properly loading when the `Configuration` object is instantiated. You could validate this theory by adding a breakpoint in the `Configuration` class upon starting the application server.

So, the solution is to provide the `hibernate-types.properties` file in such a way that JBoss can locate it from the same module that starts the Hibernate Types library.

**Comment by JensPiegsa** [comment]
Thanks for your speedy reply, Vlad. That might be the issue. If so, there is a dependency that needs to be declared in WildFly's internal Hibernate module? The properties file is one thing, the other is that the referenced class file itself must also be accessible, right?

**Comment by vladmihalcea** [comment]
> If so, there is a dependency that needs to be declared in WildFly's internal Hibernate module? 

Most likely. I only know that JBoss has its own module system, but I don't know how it works or how to configure it. 

> The properties file is one thing, the other is that the referenced class file itself must also be accessible, right?

Yes, of course. The class will be loaded at runtime using the fully-qualified class name given by the property.

**Comment by JensPiegsa** [comment]
This issue is not solved yet. However, I made some progress:

I managed to debug the deployment execution inside of the Docker composition consolidating that my class has not been instanciated, but it led me on the path that the `hibernate.types.jackson.object.mapper` property can also be provided as a Java system property. This then worked!

But the class loading / accessing issue persists at the next layer:
Deserialization requires me to access a repository (implemented as a stateless session bean) to load entities. I used JNDI to lookup the EJB instance, but invocation of any business method of that bean causes the application server to hang. 

On second thought, from a logical point of view, there is little point in introducing a directed dependency: why should / how could a system module of the application server, namely hibernate depend on the deployed application? This should work without an explicit dependency, like Hibernate also accesses my JPA entities (?).
